### PR TITLE
Fixes #12935 - Refactor jQuery methods to match version 1.11

### PR DESCRIPTION
**Pull request by shlomizadok**

### CODE REVIEW

File: app/assets/javascripts/editor.js [hunk 1]
```diff
@@ -22,11 +22,12 @@ $(document).on('change','#mode', function(){
 
 function onEditorLoad(){
   var editor_source = $(".editor_source");
-  if ($.browser && $.browser.msie && $.browser.version.slice(0,1) < 10) {
+  // IE > 11 identifies as Trident
+  if (navigator.userAgent.indexOf('MSIE') != -1) {
```
Comment (tbrisker) [inline app/assets/javascripts/editor.js:26]: Do we actually still need to support IE < 10? I tend to think not.

File: app/assets/javascripts/compute_resource.js [hunk 2]
```diff
@@ -166,7 +166,7 @@ function disable_libvirt_dropdown(item){
 
 function enable_libvirt_dropdown(item){
   item.attr("disabled",false);
```
Comment (tbrisker) [inline app/assets/javascripts/compute_resource.js]: this should also be `.prop`

File: app/assets/javascripts/lookup_keys.js [hunk 3]
```diff
@@ -157,7 +157,7 @@ function changeCheckboxEnabledStatus(checkbox, shouldEnable) {
     $(checkbox).attr('disabled', null);
   }
   else {
-    $(checkbox).attr('checked', false);
+    $(checkbox).prop('checked', false);
     $(checkbox).attr('disabled', 'disabled');
```
Comment (tbrisker) [inline app/assets/javascripts/lookup_keys.js]: this should also use `.prop`

File: app/assets/javascripts/lookup_keys.js [hunk 4]
```diff
@@ -157,7 +157,7 @@ function changeCheckboxEnabledStatus(checkbox, shouldEnable) {
     $(checkbox).attr('disabled', null);
```
Comment (tbrisker) [inline app/assets/javascripts/lookup_keys.js]: this should either use `.removeAttr('disabled')` or `.prop('disabled', false)`, not sure which is better in this case

**Comment by tbrisker** [comment]
There are quite a few `.attr(<some prop>...` you missed. 
Note this also applies to selectors (e.g. `a[disabled="disabled"]`) that need updating and to other properties besides `checked` such as `selected`, `disabled`, and more - see the jQuery 1.6.1 [release notes](http://blog.jquery.com/2011/05/12/jquery-1-6-1-released/) for a full list.

**Comment by shlomizadok** [comment]
@tbrisker - thanks for reviewing. Hopefully I covered all the `attr`s now :)

File: app/assets/javascripts/compute_resource.js [hunk 5]
```diff
@@ -94,22 +94,22 @@ function add_volume(item){
...
-  disable_element( $('[id$='+new_id+'_bootable_true]').attr('checked', item.bootable));
+  disable_element( $('[id$='+new_id+'_bootable_true]').prop('checked', item.bootable));
   $('[id$='+new_id+'_id]').val(7);
   $('[id$='+new_id+'_storage_domain]').next().hide();
 }
 
 function disable_element(element){
   element.clone().attr('type','hidden').appendTo(element);
-  element.attr('disabled', 'disabled');
+  element.prop('disabled', false);
```
Comment (tbrisker) [inline app/assets/javascripts/compute_resource.js]: this should be `true`

File: app/assets/javascripts/host_checkbox.js.erb [hunk 6]
```diff
@@ -43,7 +43,7 @@ function toggle_actions() {
   var dropdown = $("#submit_multiple a");
   if ($.foremanSelectedHosts.length == 0) {
     dropdown.addClass("disabled hide");
-    dropdown.attr('disabled', 'disabled');
+    dropdown.prop('disabled', false);
```
Comment (tbrisker) [inline app/assets/javascripts/host_checkbox.js.erb]: should be `true`

File: app/assets/javascripts/lookup_keys.js [hunk 7]
```diff
@@ -154,11 +154,11 @@ function toggleOverrideValue(item) {
...
 function changeCheckboxEnabledStatus(checkbox, shouldEnable) {
   if (shouldEnable) {
-    $(checkbox).attr('disabled', null);
+    $(checkbox).prop('disabled', false);
   }
   else {
-    $(checkbox).attr('checked', false);
-    $(checkbox).attr('disabled', 'disabled');
+    $(checkbox).prop('checked', false);
+    $(checkbox).prop('disabled', 'disabled');
```
Comment (tbrisker) [inline app/assets/javascripts/lookup_keys.js]: should be `true`

File: app/assets/javascripts/editor.js (hunk 1 above)
Comment (ares) [inline app/assets/javascripts/editor.js:26]: I think we do

File: app/assets/javascripts/compute_resource.js (hunk 5 above)
Comment (shlomizadok) [inline app/assets/javascripts/compute_resource.js]: the truth is out there :smiley:

File: app/assets/javascripts/trends.js [hunk 8]
```diff
@@ -6,10 +6,10 @@ $(function() {
 function trendTypeSelected(item){
   var is_fact = ($(item).val() == "FactName");
   var edit_mode = $(item).attr('disabled');
-  $("[id$='trend_trendable_id']").attr('disabled', (is_fact && !edit_mode) ? null : 'disabled');
-  $("[id$='trend_name']").attr('disabled', (is_fact && !edit_mode) ? null : 'disabled');
+  $("[id$='trend_trendable_id']").prop('disabled', (is_fact && !edit_mode) ? false : true);
+  $("[id$='trend_name']").attr('disabled', (is_fact && !edit_mode) ? false : true);
```
Comment (tbrisker) [inline app/assets/javascripts/trends.js]: this should also be prop

File: app/assets/javascripts/host_edit.js [hunk 9]
```diff
@@ -394,7 +394,7 @@ function medium_selected(element){
   var obj = (type == "hosts" ? "host" : "hostgroup");
   var attrs = {};
   attrs[obj] = attribute_hash(['medium_id', 'operatingsystem_id', 'architecture_id']);
-  attrs[obj]["use_image"] = $('*[id*=use_image]').attr('checked') == "checked";
+  attrs[obj]["use_image"] = $('*[id*=use_image]').is(':checked') == true;
```
Comment (tbrisker) [inline app/assets/javascripts/host_edit.js]: .is(':checked') already returns boolean, no need to compare to true.

File: app/assets/javascripts/host_edit.js [hunk 10]
```diff
@@ -411,7 +411,7 @@ function use_image_selected(element){
   var obj = (type == "hosts" ? "host" : "hostgroup");
   var attrs = {};
   attrs[obj] = attribute_hash(['medium_id', 'operatingsystem_id', 'architecture_id', 'model_id']);
-  attrs[obj]['use_image'] = ($(element).attr('checked') == "checked");
+  attrs[obj]['use_image'] = ($(element).is(':checked') == true);
```
Comment (tbrisker) [inline app/assets/javascripts/host_edit.js]: ditto

**Comment by shlomizadok** [comment]
Hope addressed all

**Comment by shlomizadok** [comment]
Test failure unrelated

File: app/assets/javascripts/taxonomy_edit.js [hunk 11]
```diff
@@ -8,7 +8,7 @@ function ignore_checked(item){
   var current_select = $(item).closest('.tab-pane').find('select');
 
   if ($(item).is(':checked')) {
-     current_select.attr('disabled', 'disabled');
+     current_select.prop('disabled', true);
   } else {
      current_select.removeAttr('disabled');
```
Comment (tbrisker) [inline app/assets/javascripts/taxonomy_edit.js:13]: This should also use `prop`. There are also a couple more occurrences of `removeAttr` for properties such as `readonly` and `selected`, please fix them as well.

**Comment by tbrisker** [comment]
@shlomizadok I'm sorry, but running a search for `.attr(` returns quite few more lines that need updating.

> The .prop() method should be used for boolean attributes/properties and for properties which do not exist in html (such as window.location). All other attributes (ones you can see in the html) can and should continue to be manipulated with the .attr() method.
> (http://blog.jquery.com/2011/05/10/jquery-1-6-1-rc-1-released/)

**Comment by shlomizadok** [comment]
Closing this... may return to it in the future :smile:

**Pull request closed without merging**